app/scripts/proactive/view/BeautifiedModalAdapter.js

```diff
@@ -30,8 +32,25 @@ define(
                 container.find("[data-help]").each(function() {
                     that.addHelpTooltip($(this))
                 })
+                // adding auto-complete
+                container.find("input").each(function() {
+                    that.addAutoComplete($(this))
+                })
 
             },
+            addAutoComplete: function(el) {
+                if (el.hasClass("jobGenericInfo")) {
+                    var configAC = config.autoCompleteJobGenericInfo;
+                    configAC.selector = function() { return el[0]};
+                    var autoCompleteJS = new autoComplete(configAC);
+                    autoCompleteJS.init();
+                } else if (el.hasClass("taskGenericInfo")) {
+                    var configAC = config.autoCompleteTaskGenericInfo;
+                    configAC.selector = function() { return el[0]};
+                    var autoCompleteJS = new autoComplete(configAC);
+                    autoCompleteJS.init();
+                }
```

Comment:
avoid duplication, better to do like this:
`
var configAC;
if (el.hasClass("jobGenericInfo")) {
  configAC = config.autoCompleteJobGenericInfo;
} else if (el.hasClass("taskGenericInfo")) {
  configAC = config.autoCompleteTaskGenericInfo;
}
configAC.selector = function() {
  return el[0]
};
var autoCompleteJS = new autoComplete(configAC);
autoCompleteJS.init()
`

Reply:
@medou-boushab it does not work because we have if / else if and not if / else.
The autocomplete object is not created and initialized all the time

Reply:
I understand, thanks